### ISSUE TITLE
Assembler: `lui` doesn't follow RiSC-16 PDF

### ISSUE THREAD

**Issue by alvaro-cuesta** [Hackchain/hackchain-core#1]
Although `lui` isn't explained very well in [RiSC-16 PDF](http://www.eng.umd.edu/~blj/RiSC/RiSC-isa.pdf), it says in page 3:

```
lui regA, immed        R[regA] <- immed & 0xffc0
```

But your assembler implementation [seems to accept only immediates up to `0x3ff`](https://github.com/Hackchain/hackchain-core/blob/master/lib/hackchain/interpreter/asm.js#L244-L246), which corresponds to the binary representation of the opcode but not the mnemonic assembly representation implied by the PDF.

As I understand the PDF, the assembler should accept `0x3000`, `0x3001`, etc. as immediates but generate the same opcode for all of them (i.e. `rA = 0xC0 << 6`). Dissassembling the opcode would always output `lui rA, 0x3000` though since there's no way to recover the original immediate from the opcode.

Both interpretations are fine, but the assembler confused me when it didn't behave as the PDF says.

**Comment by indutny** [comment]
Assembler says that:

```
 immediate (0 to 0x3FF)
```

And the action of `lui` is to put `imm << 6` to the register. The misleading `0xffc0` is actually just `0x3ff << 6`.

**Comment by indutny** [comment]
Gosh, I see what you mean. Assembler code should look like this: `lui r1, 0x8000`.

**Comment by indutny** [comment]
Should be fixed in 4.0.0, thank you!
